src/vos/tests/vts_pm.c

```diff
@@ -743,6 +743,19 @@ punch_model_test(void **state)
 	assert_int_equal(rex.rx_idx, 0);
 	assert_int_equal(rex.rx_nr, strlen(latest));
 	assert_int_equal(max_write, 11);
+
+	/** Just query max_write */
+	rc = vos_obj_query_key(arg->ctx.tc_co_hdl, oid, 0, 11, NULL, NULL, NULL, &max_write, 0, 0,
+			       NULL);
+	assert_rc_equal(rc, 0);
+	assert_int_equal(max_write, 11);
+
+	/** Invalid arguments tests */
+	rc = vos_obj_query_key(arg->ctx.tc_co_hdl, oid, DAOS_GET_MAX, 11, NULL, NULL, NULL,
+			       &max_write, 0, 0, NULL);
+	assert_rc_equal(rc, -DER_INVAL);
```

Comment:
It may be helpful to add some comment to explain the reason for INVAL: not because of non-NULL "max_write", instead, it miss the "DKEY|AKEY|RECX".